File: R2.01a/src/TP3/billetterie/Billet.java
package TP3;

public class Billet {
	private double prixAuKm;
	private final Trajet trajet;

	public Billet(Trajet trajet, double prixAuKm) {
		this.prixAuKm = prixAuKm;
		this.trajet = trajet;
	}

	public String getDepart() {
		return trajet.getDepart();
	}

	public String getArrivee() {
		return trajet.getArrivee();
	}

	public int getDistance() {
		return trajet.getDistance();
	}

	public double getPrixAuKm() {
		return prixAuKm;
	}

	public double getPrix() {
		return prixAuKm * trajet.getDistance();
	}
}

File: S3/R3.04/TP03/exercice2/Billet.cpp
#include <iostream>
#include "Billet.h"
using namespace std;

Billet::Billet(Trajet* trajet, Tarif* tarif)
{
    this->trajet = trajet;
    this->tarif = tarif;
}

float Billet::getPrix() const
{
    return  tarif->getPrixAuKm()*trajet->getDistance();
}

Tarif& Billet::getTarif() const
{
    return *tarif;
}

Trajet& Billet::getTrajet() const
{
    return *trajet;
}


